pkg/vpcmodel/abstractVPC.go

```diff
@@ -66,9 +66,14 @@ const (
 
 type Node interface {
 	VPCResourceIntf
-	Cidr() string
+	// CidrOrAddress returns the string of the Node's ip-address (for internal node) or cidr ) (for external node)
+	CidrOrAddress() string
```

Comment:
This is much better than before. Still, I would minimize the usage of this ambiguity function to the very minimal - e.g. printing functions.
Perhaps restore the function `Cidr()` and have it return an empty string for internal `Nodes`? And then use each `Cidr(), IPBlock()` in the relevant context and use `CidrOrAddress()` only when it can indeed be both

Reply:
Indeed I tried minimize the usage of this function.

I think it might be confusing/error-prone to have a `Cidr()` function that sometimes returns an empty string. 
The `IPBlock()` function is not a replacement for `CidrOrAddress()`, it does not return an IP address string, just an IPBlock object.

Reply:
To my understanding, this function is used when we need really need is `Cidr`. Personally I find it confusing and even error prune.
We are calling this function almost 50 times, which is too much, I think. 
Seems there is something we are not doing right here. Should we open a follow up issue? at the very minimal I would leave a README

Reply:
> this function is used when we need really need is Cidr

Not necessarily, for internal nodes we sometimes really need an address.

> We are calling this function almost 50 times

Less. these are the places where this is called (after the follow up PR that also removes some calls):
```
pkg/ibmvpc/nacl_analysis.go:    return nil, fmt.Errorf(notFoundMsg, isIngress, targetNode.CidrOrAddress(), subnet.cidr, nodeInSubnet.CidrOrAddress())
pkg/ibmvpc/nacl_analysis.go:    return nil, nil, fmt.Errorf(notFoundMsg, isIngress, targetNode.CidrOrAddress(), subnet.cidr, nodeInSubnet.CidrOrAddress())
pkg/ibmvpc/parser.go:           fmt.Println(strings.Join([]string{n.Kind(), n.CidrOrAddress(), n.Name(), n.UID()}, separator))
pkg/ibmvpc/vpc.go:      return member.CidrOrAddress(), target.IPBlock()
pkg/vpcmodel/drawioGenerator.go:                tooltip = append(tooltip, n.(*ExternalNetwork).CidrOrAddress())
pkg/vpcmodel/externalNetwork.go:                ipList[i] = n.CidrOrAddress()
pkg/vpcmodel/grouping.go:               cidrList[i] = n.CidrOrAddress()
pkg/vpcmodel/nodesConnectivity.go:                      srcName := srcNode.CidrOrAddress()
pkg/vpcmodel/nodesConnectivity.go:                      dstName := dstNode.CidrOrAddress()
pkg/vpcmodel/nodesConnectivity.go:                      strList = append(strList, getConnectionStr(peerNode.CidrOrAddress(), node.CidrOrAddress(), conn.String(), " [inbound]"))
pkg/vpcmodel/nodesConnectivity.go:                      strList = append(strList, getConnectionStr(node.CidrOrAddress(), peerNode.CidrOrAddress(), conn.String(), " [outbound]"))
pkg/vpcmodel/nodesConnectivity.go:                      strList = append(strList, getConnectionStr(src.(Node).CidrOrAddress(), dst.(Node).CidrOrAddress(), conns.String(), ""))
pkg/vpcmodel/semanticDiff.go:                   nodeSameCidr := findNodeWithCidr(other.Nodes, ep.(Node).CidrOrAddress())
pkg/vpcmodel/semanticDiff.go:           if node.CidrOrAddress() == cidr {
```

> Should we open a follow up issue

Yes, maybe have a follow up issue, to see if/how this can be improved.

Reply:
opened #397 and assigned both of us for now